compiler/optimizer/LocalOpts.cpp

```diff
@@ -8367,30 +8367,16 @@ TR_ColdBlockOutlining::reorderColdBlocks()
       {
       TR::Block *currentBlock = currentTree->getNode()->getBlock();
 
-      // PLX: do not separate block with internal call
-      // and the next block. The same for ASM block
-#if 0
-      if (currentBlock->isGenAsmBlock() &&
-          currentBlock->getNextBlock())
-         {
-         // KP: Make sure we don't skip the block that was originally the last.
-         if (currentBlock->getExit() == lastTree) break;
-         currentBlock = currentBlock->getNextBlock();
-         if (currentBlock->getExit() == lastTree) break;
-         currentBlock = currentBlock->getNextBlock();
-         }
-#endif
-
       if (currentBlock == NULL) break;
       exitTree = currentBlock->getExit();
 
       if (exitTree == lastTree) break;
 
       // have to check if a getNextTreeTop still exists as the current block may have been swung down to be the new last block
       // For PLX, we have to move user marked cold block
-      //if (exitTree->getNextTreeTop() && exitTree->getNextTreeTop()->getNode()->getBlock()->isGenAsmBlock())
+      //if (exitTree->getNextTreeTop())
 #if 0
-      if ( exitTree->getNextTreeTop() && exitTree->getNextTreeTop()->getNode()->getBlock()->isGenAsmBlock())
+      if ( exitTree->getNextTreeTop())
```

Comment:
I have taken a look at all the changes and made sure the logic hasn't been changed. Should be safe to merge now.